core/src/banking_stage/transaction_scheduler/thread_aware_account_locks.rs

```diff
@@ -28,20 +28,25 @@ struct AccountReadLocks {
     lock_counts: [LockCount; MAX_THREADS],
 }
 
+#[derive(Default)]
+struct AccountLocks {
+    pub write_locks: Option<AccountWriteLocks>,
+    pub read_locks: Option<AccountReadLocks>,
+}
+
 /// Thread-aware account locks which allows for scheduling on threads
 /// that already hold locks on the account. This is useful for allowing
 /// queued transactions to be scheduled on a thread while the transaction
 /// is still being executed on the thread.
 pub(crate) struct ThreadAwareAccountLocks {
     /// Number of threads.
     num_threads: usize, // 0..MAX_THREADS
-    /// Write locks - only one thread can hold a write lock at a time.
-    /// Contains how many write locks are held by the thread.
-    write_locks: HashMap<Pubkey, AccountWriteLocks>,
-    /// Read locks - multiple threads can hold a read lock at a time.
-    /// Contains thread-set for easily checking which threads are scheduled.
-    /// Contains how many read locks are held by each thread.
-    read_locks: HashMap<Pubkey, AccountReadLocks>,
+    /// Account locks.
+    /// Write Locks - only one thread can hold a write lock at a time.
+    ///     Contains how many write locks are held by the thread.
+    /// Read Locks - multiple threads can hold a read lock at a time.
+    ///     Contains thread-set for easily checking which threads are scheduled.
```

Comment:
nit: better off moving these doc comments to the shiny `AccountLocks`?

Reply:
56962014d3d6c7d70d5d0ed144dc16a5a8a44ab1